src/lambda/commands/importLambda.ts

```diff
@@ -0,0 +1,233 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as AdmZip from 'adm-zip'
+import { Lambda } from 'aws-sdk'
+import * as fs from 'fs-extra'
+import got from 'got'
+import * as _ from 'lodash'
+import * as path from 'path'
+import { Stream } from 'stream'
+import { promisify } from 'util'
+import * as vscode from 'vscode'
+import { LaunchConfiguration, getReferencedHandlerPaths } from '../../shared/debug/launchConfiguration'
+import { ext } from '../../shared/extensionGlobals'
+import { makeTemporaryToolkitFolder, fileExists } from '../../shared/filesystemUtilities'
+import { getLogger } from '../../shared/logger'
+import { createCodeAwsSamDebugConfig } from '../../shared/sam/debugger/awsSamDebugConfiguration'
+import { ExtensionDisposableFiles } from '../../shared/utilities/disposableFiles'
+import * as pathutils from '../../shared/utilities/pathUtils'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { Window } from '../../shared/vscode/window'
+import { LambdaFunctionNode } from '../explorer/lambdaFunctionNode'
+import { getFamily, RuntimeFamily } from '../models/samLambdaRuntime'
+import { showConfirmationMessage } from '../../s3/util/messages'
+import { addWorkspaceFolder } from '../../shared/utilities/workspaceUtils'
+
+const pipeline = promisify(Stream.pipeline)
+
+export async function importLambdaCommand(functionNode: LambdaFunctionNode, window = Window.vscode()) {
```

Comment:
1. function accepts `window` but still makes direct vscode calls
2. this signature makes me believe the function is set up for testing but has no test coverage